### PR TITLE
For #4124: Migrate BrowserToolbar to Libstate

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserStore.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package org.mozilla.fenix.components.toolbar
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserStore.kt]: license header missing.

File: app/src/main/java/org/mozilla/fenix/search/SearchInteractor.kt [hunk 2]
```diff
@@ -18,7 +18,7 @@ import org.mozilla.fenix.search.toolbar.ToolbarInteractor
 
 /**
  * Interactor for the search screen
- * Provides implementations for the AwesomeBarView and ToolbarView
+ * Provides implementations for the AwesomeBarView and BrowserToolbarView??? <- TODO?
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/search/SearchInteractor.kt]: nit: Is this still a TODO?

File: app/src/main/java/org/mozilla/fenix/search/toolbar/ToolbarView.kt [hunk 3]
```diff
@@ -21,24 +21,24 @@ import org.mozilla.fenix.search.SearchState
...
-     * Called when a user hits the return key while ToolbarView has focus.
-     * @param url the text inside the ToolbarView when committed
+     * Called when a user hits the return key while BrowserToolbarView has focus.
+     * @param url the text inside the BrowserToolbarView when committed
      */
     fun onUrlCommitted(url: String)
 
     /**
-     * Called when a removes focus from the ToolbarView
+     * Called when a removes focus from the BrowserToolbarView
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/search/toolbar/ToolbarView.kt]: nit: [BrowserToolbarView] to link to it.

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 4]
```diff
@@ -382,46 +413,23 @@ class BrowserFragment : Fragment(), BackHandler {
...
-                bookmarkTapped = {
-                    lifecycleScope.launch { bookmarkTapped(it) }
-                },
-                appLinksUseCases = requireComponents.useCases.appLinksUseCases
-            )
-        )
-
-        consumeFrom(quickActionSheetStore) {
-            quickActionSheetView.update(it)
+        browserStore.observe(view) {
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: `consumeFrom` was right here so we can replace this now with:
```kotlin
consumeFrom(browserStore) {
  quickActionSheetView.update(it)
  browserToolbarView.update(it)
}
```

See also: https://github.com/mozilla-mobile/fenix/pull/4294

Comment (csadilek) [reply app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: Thanks for adding test! Let's just address ^^ and squash/land this. Did another round of testing too!

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: Resolved this 😄

**Pull request merged**